File: 0775. Global and Local Inversions/Solution.cpp
#include <algorithm>
#include <vector>
using namespace std;

class Solution {
public:
    bool isIdealPermutation_MK1(vector<int>& nums)
    {
        int cmax = 0, n = nums.size();
        for (int i = 0; i < n - 2; i++) {
            cmax = max(cmax, nums[i]);
            if (cmax > nums[i + 2])
                return false;
        }
        return true;
    }

    bool isIdealPermutation_MK2(vector<int>& nums)
    {
        for (int i = 0; i < nums.size(); i++)
            if (abs(nums[i] - i) > 1)
                return false;
        return true;
    }
};

File: 0775. Global and Local Inversions/Solution.py
from typing import List


class Solution:
    def isIdealPermutation_MK1(self, nums: List[int]) -> bool:
        cmax = 0
        for i, e in enumerate(nums[:-2]):
            cmax = max(cmax, e)
            if cmax > nums[i + 2]:
                return False
        return True

    def isIdealPermutation_MK2(self, nums: List[int]) -> bool:
        return all(abs(i - e) <= 1 for i, e in enumerate(nums))
